### PR TITLE
RELATED: BB-1326 Pass executionResponse to header predicate calls

### PR DESCRIPTION
**Pull request by petrjanu**
This PR adds ability to work with execution response in header predicates. Execution response is needed to fully support drilling on AMs because only AFM is not enough to obtain information such identifier from measure headers.

### CODE REVIEW

**Comment by petrjanu** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2460/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1863/ : SUCCESS in 2m 06s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/982/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2461/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/574/ : UNSTABLE in 14m 52s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2462/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1864/ : SUCCESS in 1m 47s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/983/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2463/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/575/ : SUCCESS in 13m 10s

**Comment by petrjanu** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-696.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2464/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/576/ : SUCCESS in 18m 10s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2467/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1866/ : SUCCESS in 3m 44s

File: src/factory/tests/mocks.ts [hunk 1]
```diff
@@ -194,3 +215,43 @@ export const amMeasureHeaderItems = {
         }
     }
 };
+
+export const executionResultWithAmMeasures: Execution.IExecutionResponse = {
```
Comment (tomassatinsky) [inline src/factory/tests/mocks.ts]: "result" in name but "response" type. Seems inconsistent also with `emptyExecutionResponse`

File: src/components/visualizations/chart/chartOptionsBuilder.ts [hunk 2]
```diff
@@ -891,7 +901,7 @@ export function findAttributeInDimension(
         });
 }
 
-export function getDrillContext(stackByItem: any, viewByItem: any, measures: AFM.IMeasure[], afm: AFM.IAfm) {
+export function getDrillContext(stackByItem: any, viewByItem: any, measures: any[], afm: AFM.IAfm) {
```
Comment (tomassatinsky) [inline src/components/visualizations/chart/chartOptionsBuilder.ts]: why `measures: AFM.IMeasure[]` is now `any`?

File: src/components/visualizations/chart/chartOptionsBuilder.ts [hunk 3]
```diff
@@ -871,6 +873,14 @@ export function findMeasureGroupInDimensions(dimensions: Execution.IResultDimens
         });
 }
 
+export function findMeasureGroup(
```
Comment (tomassatinsky) [inline src/components/visualizations/chart/chartOptionsBuilder.ts]: This function seems either superfluous or not so well named to me. I'd suggest 
a) use `findMeasureGroupInDimensions(executionResponse.dimensions)`
b) name it better, for instance `findMeasureGroupInResponse`
I prefer `a)`

File: src/components/visualizations/chart/chartOptionsBuilder.ts (hunk 2 above)
Comment (tomassatinsky) [reply src/components/visualizations/chart/chartOptionsBuilder.ts]: ignore that, it see the (typing) mess around

**Comment by tomassatinsky** [comment]
Seems OK to me apart from the one small thing I commented

File: src/components/visualizations/chart/chartOptionsBuilder.ts (hunk 2 above)
Comment (petrjanu) [reply src/components/visualizations/chart/chartOptionsBuilder.ts]: Better to put `any` than wrong type, that was used before and due to obfuscated TS was invisible to TSC... Needs refactor. Wont fix in this PR.

File: src/components/visualizations/chart/chartOptionsBuilder.ts (hunk 3 above)
Comment (petrjanu) [reply src/components/visualizations/chart/chartOptionsBuilder.ts]: Did a)

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2496/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1894/ : FAILURE in 56s

**Comment by petrjanu** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2497/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1895/ : SUCCESS in 5m 57s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1002/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2498/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/587/ : SUCCESS in 15m 46s

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/532/ : SUCCESS in 14s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1896/ : SUCCESS in 11m 06s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1004/ : UNSTABLE in 24m 21s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/490/ : SUCCESS in 22m 30s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/534/ : SUCCESS in 18s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1898/ : SUCCESS in 2m 09s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1006/ : SUCCESS in 15m 53s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/492/ : SUCCESS in 14m 28s

**Pull request merged**